write-fonts/src/tables/variations/ivs_builder.rs

```diff
@@ -83,28 +99,64 @@ impl VariationStoreBuilder {
         Encoder::new(&self.delta_sets, self.all_regions.len() as u16)
     }
 
+    /// Build the `ItemVariationStore` table
+    ///
+    /// This also returns a structure that can be used to remap the temporarily
+    /// assigned delta set Ids to their final `VariationIndex` values.
     pub fn build(self) -> (ItemVariationStore, VariationIndexRemapping) {
         let mut key_map = VariationIndexRemapping::default();
-        let mut encoder = self.encoder();
-        encoder.optimize();
-        let subtables = encoder.encode(&mut key_map);
+        let subtables = if matches!(self.delta_sets, DeltaSetStorage::Direct(_)) {
+            vec![self.build_unoptimized(&mut key_map)]
+        } else {
+            let mut encoder = self.encoder();
+            encoder.optimize();
+            encoder.encode(&mut key_map)
+        };
         let region_list = self.make_region_list();
         (ItemVariationStore::new(region_list, subtables), key_map)
     }
+
+    /// Build a single ItemVariationData subtable
+    fn build_unoptimized(
+        &self,
+        key_map: &mut VariationIndexRemapping,
+    ) -> Option<ItemVariationData> {
+        // first pick an encoding capable of representing all items:
+        let n_regions = self.all_regions.len() as u16;
+        let mut shape = RowShape(vec![ColumnBits::None; n_regions as usize]);
+        let mut temp = RowShape::default();
+
+        for (delta, _) in self.delta_sets.iter() {
+            temp.reuse(delta, n_regions);
+            if !shape.can_cover(&temp) {
+                shape = shape.merge(&temp);
+            }
+        }
+
+        // then encode everything with that encoding.
+        let encoding = Encoding {
+            shape,
+            deltas: self.delta_sets.iter().collect(),
+        };
+        debug_assert!(
+            encoding.deltas.len() <= u16::MAX as usize,
+            "unmapped variation store supports at most u16::MAX items"
+        );
+        encoding.encode(key_map, 0)
```

Comment:
should we assert as sanity-check that `key_map` only contains an identity mapping so that gid:varidx relationship holds for the unoptimized builder?

Reply:
my expectation is that the key map isn't used in this case, but can certainly add this to the test, we do set the values.